match/tensordata.py

```diff
@@ -149,14 +153,61 @@ def __reshape(self, shape: tuple):
         # The strides change when the shape does, so they must be reinitialized.
         self.__initialize_strides()
 
-    def reshape(self, *shape: int) -> 'TensorData':
+    def reshape(self, *shape: int) -> "TensorData":
```

Comment:
```suggestion
    def reshape(self, *shape: int) -> TensorData:
```

I think we should move to Python 3.12 unless we have a reason not to.